contracts/protocols/mainnet/morpho/aave_v2/helpers.sol

```diff
@@ -0,0 +1,244 @@
+        bool isLiquidatable;
+        uint256 liquidationThreshold;
+        UserMarketData[] marketData;
+    }
+
+    IAaveLens internal aavelens = IAaveLens(0x507fA343d0A90786d86C7cd885f5C49263A91FF4);
+    IMorpho internal aaveMorpho = IMorpho(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0);
+    IAave internal protocolData = IAave(getAaveProtocolDataProvider());
+    IAave internal incentiveData = IAave(getAaveIncentivesController());
+
+    function getLiquidatyData(
+        MarketDetail memory marketData_,
+        address poolTokenAddress_,
+        address asset
+    ) internal view returns (MarketDetail memory) {
+        (
+            ,
+            ,
+            ,
+            ,
+            ,
+            ,
+            ,
+            marketData_.aaveData.ltv,
+            marketData_.aaveData.liquidationThreshold,
+            marketData_.aaveData.liquidationBonus,
+
+        ) = aavelens.getMarketConfiguration(poolTokenAddress_);
+
+        (, , address vToken_) = protocolData.getReserveTokensAddresses(asset);
+
+        (, marketData_.aaveData.aEmissionPerSecond, ) = incentiveData.getAssetData(asset);
+        (, marketData_.aaveData.vEmissionPerSecond, ) = incentiveData.getAssetData(vToken_);
+        (marketData_.aaveData.availableLiquidity, , , marketData_.aaveData.liquidityRate, , , , , , ) = protocolData
+            .getReserveData(asset);
+        return marketData_;
+    }
+
+    function getAaveMarketData(MarketDetail memory marketData_, address poolTokenAddress_)
+        internal
+        view
+        returns (MarketDetail memory)
+    {
+        marketData_.config.poolTokenAddress = poolTokenAddress_;
+        (
+            marketData_.config.underlyingToken,
+            ,
+            marketData_.flags.isP2PDisabled,
+            marketData_.flags.isPaused,
+            marketData_.flags.isPartiallyPaused,
+            marketData_.reserveFactor,
+            ,
+            ,
+            ,
+            ,
+            marketData_.config.decimals
+        ) = aavelens.getMarketConfiguration(poolTokenAddress_);
+
+        marketData_ = getLiquidatyData(marketData_, poolTokenAddress_, marketData_.config.underlyingToken);
+
+        return marketData_;
+    }
+
+    function getMarketData(address poolTokenAddress) internal view returns (MarketDetail memory marketData_) {
+        marketData_ = getAaveMarketData(marketData_, poolTokenAddress);
+
+        (
+            marketData_.avgSupplyRate,
+            marketData_.avgBorrowRate,
+            marketData_.totalP2PSupply,
+            marketData_.totalP2PBorrows,
+            marketData_.totalPoolSupply,
+            marketData_.totalPoolBorrows
+        ) = aavelens.getMainMarketData(poolTokenAddress);
+
+        (
+            marketData_.p2pSupplyRate,
+            marketData_.p2pBorrowRate,
+            marketData_.poolSupplyRate,
+            marketData_.poolBorrowRate
+        ) = aavelens.getRatesPerYear(poolTokenAddress);
+
+        (
+            ,
```

Comment:
@Richa-iitr why are you filtering data? Why not just return all the data from lens?

Reply:
Updated.